syntax = "proto3" ;

service Booking {
    rpc GetBookingsByUserId(UserId) returns (stream BookingData) {}
    rpc checkBookingValidity(MovieSchedule) returns (Validity) {}
    rpc addBooking(MovieScheduleWithUserId) returns (MovieScheduleWithUserId) {}
    rpc deleteBooking(MovieScheduleWithUserId) returns (MovieScheduleWithUserId) {}
}

message UserId {
    string userid = 1;
}

message BookingData {
    string date = 1;
    repeated string movies = 2;
}

message Validity {
    bool validity = 1;
}

message MovieScheduleWithUserId {
    UserId userid = 1;
    MovieSchedule movieSchedule = 2;
}

message MovieSchedule {
    string date = 1;
    string movieid = 2;
}